src/web/demo/src/main/java/org/geoserver/web/demo/PreviewLayer.java

```diff
@@ -339,10 +339,9 @@ private String findFeatureTypeGmlVersion(AttributeType featureType) {
      * @param serviceName "WFS" or "WMS"
      */
     public boolean hasServiceSupport(String serviceName) {
-        LayerInfo linfo = GeoServerApplication.get().getCatalog().getLayerByName(this.getName());
-        if (linfo != null && linfo.getResource() != null && serviceName != null) {
+        if (layerInfo != null && layerInfo.getResource() != null && serviceName != null) {
```

Comment:
This looks correct, I am not sure why the previous implementation wanted to do a new layer lookup?

Reply:
Refactor happened last year, this bit of code was moved from another class that did not have the layerInfo field, but just the name.